File: app/src/main/java/com/prideven/android/HungryEats/MainFragment.java
package com.prideven.android.hungryeats;

import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.gson.Gson;
import com.prideven.android.hungryeats.databinding.RestaurantListFileBinding;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class MainFragment extends Fragment {

    private com.prideven.android.hungryeats.RestaurantViewModel mViewModel;
    private RestaurantListFileBinding mBinding;
    private RecyclerView rv;
    com.prideven.android.hungryeats.RestaurantAdapter ra;

    public static MainFragment newInstance() {
        return new MainFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {

        mBinding = DataBindingUtil.inflate(inflater, R.layout.restaurant_list_file, container, false);

        RecyclerView rv = mBinding.restaurantList;
        List<EatsRestaurantsResponseItem> items = new ArrayList<>();
        ra = new com.prideven.android.hungryeats.RestaurantAdapter(items);
        rv.setAdapter(ra);
        rv.setLayoutManager(new LinearLayoutManager(requireContext()));
        return mBinding.getRoot();

    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mViewModel = new ViewModelProvider(this).get(com.prideven.android.hungryeats.RestaurantViewModel.class);
        CharSequence zipCode = mBinding.searchZip.getQuery();
        //mViewModel.callRestaurantDataRepo(Integer.parseInt(zipCode.toString()));
        mViewModel.callRestaurantDataRepo(37.42274,-122.139956);


        mViewModel.getRestaurantDetails().observe(getViewLifecycleOwner(), listOfRestaurants ->{
            ra.setData(listOfRestaurants);
            // Update the UI.
        });

        // TODO: Use the ViewModel
    }


}

File: app/src/main/java/com/prideven/android/HungryEats/MainFragment.kt
package com.prideven.android.hungryeats

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.prideven.android.hungryeats.databinding.RestaurantListFileBinding
import java.util.*

class MainFragment : Fragment() {
    private var mViewModel: RestaurantViewModel? = null
    private var mBinding: RestaurantListFileBinding? =
        null
    private val rv: RecyclerView? = null
    var ra: RestaurantAdapter? = null
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        mBinding =
            DataBindingUtil.inflate(inflater, R.layout.restaurant_list_file, container, false)
        val rv = mBinding.restaurantList
        val items: List<EatsRestaurantsResponseItem> =
            ArrayList()
        ra = RestaurantAdapter(items)
        rv.adapter = ra
        rv.layoutManager = LinearLayoutManager(requireContext())
        return mBinding.getRoot()
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        mViewModel =
            ViewModelProvider(this).get(RestaurantViewModel::class.java)
        val zipCode: CharSequence = mBinding.searchZip.getQuery()
        //mViewModel.callRestaurantDataRepo(Integer.parseInt(zipCode.toString()));
        mViewModel!!.callRestaurantDataRepo(37.42274, -122.139956)
        mViewModel!!.restaurantDetails.observe(
            viewLifecycleOwner,
            Observer { listOfRestaurants: List<EatsRestaurantsResponseItem?>? ->
                ra!!.setData(
                    listOfRestaurants
                )
            }
        )

        // TODO: Use the ViewModel
    }

    companion object {
        fun newInstance(): MainFragment {
            return MainFragment()
        }
    }
}